src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -274,6 +274,9 @@ ss::future<ss::stop_iteration> leader_balancer::balance() {
      */
     std::optional<model::ntp> ntp;
     for (const auto& topic : _topics.topics_map()) {
+        if (topic.second.is_topic_replicated()) {
+            continue;
+        }
```

Comment:
I was thinking if the configuration was some other variant type, that would force the user to call `std::visit` to handle all cases. Instead of the current method which is I hope you call `is_topic_replicated` before attempting to grab either the source topic or revision, otherwise it would crash.

Its would be akin to a language with ASTs where one would use type constructors and pattern matching to unwrap. Didn't know if that would be too much here, since there's only one other use of `std::visit` in the entire codebase. Opinions?

Reply:
yeh there may not be a good solution other than the change you made. this is an example of where i was mentioning that it is concerning to me that a materialized topic carries around information like raft group id. i think this highlights the need to add assertions on those accessors. i think there isn't one currently on the topic metadata assignments.

Reply:
I'll add those assertions +1

Reply:
So currently the public accessible field `assignments` contains a raft group id which for `non_replicable_topics` this does not make sense.